File: src/main/scala/a_lists/P01.scala
package a_lists

import java.util.NoSuchElementException

// P01. Find the last element of a list
object P01 {

  def last[A](xs: List[A]): A = xs.last

  def lastByLength[A](xs: List[A]): A = xs(xs.length-1)

  def lastRecursive[A](xs: List[A]): A = xs match {
    case x :: Nil  => x
    case _ :: tail => lastRecursive(tail)
    case _         => throw new NoSuchElementException
  }

}

File: ruby/a_lists/p01.rb
class P01

  # Find the last element in a list

  def initialize(list)
    @list = list || []
  end

  def last_in_list
    @list.last
  end

  def last_by_index
    @list[-1]
  end

  def last_by_length_index
    @list[@list.length - 1]
  end

end